myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/constraints/LikeConstraint.java

```diff
@@ -0,0 +1,94 @@
+package com.ebay.myriad.scheduler.constraints;
+
+import com.google.gson.Gson;
+import java.util.Collection;
+import java.util.regex.Pattern;
+import org.apache.mesos.Protos.Attribute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Constraint for LIKE operator.
+ * Format: <mesos_slave_attribute|hostname> LIKE <regex_value>
+ */
+public class LikeConstraint implements Constraint {
+  private static final Logger LOGGER = LoggerFactory.getLogger(LikeConstraint.class);
+
+  private String lhs;
+  private String rhsRegex;
+
+  public LikeConstraint(String lhs, String rhsRegex) {
+    this.lhs = lhs;
+    this.rhsRegex = rhsRegex;
+  }
+
+  public boolean isConstraintOnHostName() {
+    return lhs.equalsIgnoreCase("hostname");
+  }
+
+  public boolean matchesHostName(String hostname) {
+    return lhs.equalsIgnoreCase("hostname") && hostname != null && Pattern.matches(rhsRegex, hostname);
```

Comment:
Let's compile regex upfront in constructor - otherwise it will compile it every time here